api/src/pcapi/core/search/__init__.py

```diff
@@ -628,3 +671,43 @@ def unindex_all_venues() -> None:
         if settings.IS_RUNNING_TESTS:
             raise
         logger.exception("Could not unindex all venues")
+
+
+def reindex_offers_if_ean_booked_since(offer_ids: list, since: datetime.datetime) -> None:
+    """
+    This function loads all offers booked the last x hours and
+    if they have an ean, it will reindex all related offers with the same ean
+    """
+
+    last_x_hours_bookings_with_ean = (
+        bookings_models.Booking.query.filter(
+            bookings_models.Booking.dateCreated >= since,
+            bookings_models.Booking.status != bookings_models.BookingStatus.CANCELLED,
+        )
+        .join(bookings_models.Booking.stock)
+        .join(offers_models.Stock.offer)
+        .join(offers_models.Offer.product)
+        .filter(sa.or_(offers_models.Offer.extraData["ean"].isnot(None)))
+        .with_entities(offers_models.Offer.extraData["ean"].label("ean"))
+    )
+
+    eans = set()
+    for booking in last_x_hours_bookings_with_ean:
+        eans.add(booking.ean)
```

Comment:
[détails / style] En plus concis, mais qui reste lisible : 

```suggestion
    eans = {booking.ean for booking in last_x_hours_bookings_with_ean}
```

C'est dans le commit "add command".